### PR TITLE
Added toggle-all functionality to the FormBlock

### PR DESCRIPTION
**Pull request by deanhannigan**
## Description

All fields in the `FormBlock` can now be toggled at the same time.

- There was a small adjustment to the design as sections are now collapsible 
~~A new `settings` context was added to the `ComponentSettingsSection` where save requests can be tracked. This can be used by any setting type to allow custom behaviour when updating. In this instance, the fields are briefly disabled while the toggle request is processed.~~

## Addresses
- https://github.com/Budibase/budibase/issues/11741

## Feature branch env
[Feature Branch Link](http://fb-feature-toggle-all-formblock-fields.fb.qa.budibase.net)

### CODE REVIEW

File: packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldConfiguration.svelte [hunk 1]
```diff
@@ -43,6 +49,10 @@
     cachedValue = cloneDeep(value)
   }
 
+  $: if (typeof $updates.resp == "string") {
```
Comment (melohagan) [inline packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldConfiguration.svelte:52]: NAB: How would this differ from:

```js
if ($updates.resp) {
  updating = false
}
```

**Review by melohagan — APPROVED** [review]
LGTM - handy feature! 😄 

I was wondering why `CellEditor` and `ColumnDrawer` have been removed?

**Comment by deanhannigan** [comment]
> LGTM - handy feature! 😄
> 
> I was wondering why `CellEditor` and `ColumnDrawer` have been removed?

Hey Mel, they were a hangover from the previous FieldConfiguration UX and they are no longer being used anywhere.

**Comment by melohagan** [comment]
> > LGTM - handy feature! 😄
> > I was wondering why `CellEditor` and `ColumnDrawer` have been removed?
> 
> Hey Mel, they were a hangover from the previous FieldConfiguration UX and they are no longer being used anywhere.

Cool, just double checking 👍

File: packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldConfiguration.svelte (hunk 1 above)
Comment (deanhannigan) [reply packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldConfiguration.svelte:52]: They achieve the same thing, but the `typeof` just communicates the contains of `resp` better

**Comment by aptkingston** [comment]
Looks great! Just was wondering around the new settings context and the flashing disabled state - what was the motivation for adding this? Disabling things while changes are in progress isn't a pattern we use elsewhere in settings, and I'm just not sure if the additional complexity is worth it. It also looks a little bit flaky since it's just checking for the existence of a string value to be able to tell when the transaction completes.

Basically I think I'd just prefer to leave out the disabled state stuff unless there's a good reason to add it, which there very well might be!

**Comment by deanhannigan** [comment]
@aptkingston I added it after testing the toggle-all in slow mode in chrome. As the toggle-all request was pending I was still  able to drag fields around and edit them. As the requests completed, the fields began jumping around and I thought it may be a bit confusing. Bit of an extreme use-case but, as you say though, it's not a current patten so I'll revert the changes. 👍

File: packages/builder/src/components/design/settings/controls/EditComponentPopover.svelte [hunk 2]
```diff
@@ -90,6 +91,7 @@
       open = true
     }
   }}
+  {disabled}
```
Comment (aptkingston) [inline packages/builder/src/components/design/settings/controls/EditComponentPopover.svelte:94]: Think we can remove this disabled prop now.

File: packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldConfiguration.svelte [hunk 3]
```diff
@@ -145,16 +148,35 @@
...
 <div class="field-configuration">
+  <div class="toggle-all">
+    <span />
+    <Toggle
+      on:change={() => {
+        let update = fieldList.map(field => ({
+          ...field,
+          active: selectAll,
+        }))
+        toggleAll(update)
```
Comment (aptkingston) [inline packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldConfiguration.svelte:170]: We should either remove the `toggleAll` function now (since its a straight pass-through to `listUpdate`) or better yet in my opinion, move the inline logic here into toggle all and keep it.

File: packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldSetting.svelte [hunk 4]
```diff
@@ -49,6 +50,7 @@
       {bindings}
       {parseSettings}
       on:change
+      disabled={updating}
```
Comment (aptkingston) [inline packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldSetting.svelte:53]: We can remove the `disabled` prop now since we can also remove this `updating` prop.

File: packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldSetting.svelte [hunk 5]
```diff
@@ -58,7 +60,13 @@
     <div class="field-label">{readableText}</div>
   </div>
   <div class="list-item-right">
-    <Toggle on:change={onToggle(item)} text="" value={item.active} thin />
+    <Toggle
+      on:change={onToggle(item)}
+      text=""
+      value={item.active}
+      thin
+      disabled={updating}
```
Comment (aptkingston) [inline packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldSetting.svelte:68]: We can remove this `disabled` prop.

**Review by aptkingston — APPROVED** [review]
LGTM!

All comments are just leftover things that can be removed now, but no blockers. Moving this to QA since its functionally complete, but would be good to do that little bit of clean up 👍
